packages/manager/apps/catalog-revamp/src/hooks/useCatalog.ts

```diff
@@ -0,0 +1,67 @@
+import { useState, useEffect } from 'react';
+import { useQuery, UseQueryResult } from '@tanstack/react-query';
+import {
+  getManagerHubCatalogList,
+  getManagerHubCatalogListQueryKey,
+} from '@/api';
+import {
+  groupItems,
+  filterByEnum,
+  getAvailableCategories,
+  matchCategories,
+  matchUniverses,
+  matchSearchText,
+  filterItems,
+  getUniverses,
+} from '@/utils/utils';
+
+export type Product = {
+  id: number;
+  name: string;
+  description: string;
+  lang: string;
+  categories: string[];
+  url: string;
+  regionTags: string[];
+  productName: string;
+  order: string;
+  universe: string;
+  category: string;
+};
+
+interface ServiceData {
+  status: number;
+  data: any;
+}
+
+interface CatalogProps {
+  categories?: string[];
+  universes?: string[];
+  searchText?: string;
+}
+
+export const useCatalog = ({
+  categories,
+  universes,
+  searchText,
+}: CatalogProps = {}) => {
+  const [products, setProducts] = useState([]);
+
+  const service: UseQueryResult<ServiceData> = useQuery({
+    queryKey: getManagerHubCatalogListQueryKey,
+    queryFn: () => getManagerHubCatalogList(),
+    staleTime: Infinity,
+  });
+
+  useEffect(() => {
+    if (service.isSuccess && service.data.status === 200) {
+      setProducts(service.data?.data?.catalog?.data || []);
+    }
+  }, [service.data]);
```

Comment:
Isn't service already a state ?
You could just return `{product: service.data?.data?.catalog?.data || []}` ?